### ISSUE TITLE
Rand256 - Problems with Zone and Spot empty coordinates.

### ISSUE THREAD

**Issue by bkueppers** [sca075/valetudo_vacuum_camera#109]
### Checklist

- [X] I have updated the integration to the latest version available
- [X] I have checked if the problem is already reported

### The problem

The integration is apparently not able to parse the map data sent out over MQTT. In the logs I find this:

```
roborock_1 : Error in get_image_from_json: list index out of range
Traceback (most recent call last):
  File "/config/custom_components/valetudo_vacuum_camera/valetudo/rand256/image_handler.py", line 431, in get_image_from_rrm
    self.room_propriety = await self.get_rooms_attributes(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/valetudo_vacuum_camera/valetudo/rand256/image_handler.py", line 586, in get_rooms_attributes
    self.room_propriety = self.extract_room_properties(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/valetudo_vacuum_camera/valetudo/rand256/image_handler.py", line 261, in extract_room_properties
    coordinates[0].pop()
    ~~~~~~~~~~~^^^
IndexError: list index out of range
```

There is mapdata coming in over MQTT at topic valetudo/roborock_1/map_data, but it does not seem to be JSON:

```
��`e��Pڂ�?>�@@Aq�WTPd��K.7�w�+*�*�?��(6��(�ʒ![PQ�  \8��_r�MҴM��ӷI�\r���>︋(�1t~�����׭����<������`^#C�����`��)Zy]+P��IDM�ANNn����o}���ZCXKn^N^��X��T;�5��F�⻫�����j��V���]^�T+(���4�n|�_�SM�f�'�'�T^�.O��HO9�H����/試�y�%zAAyAG�O�T+�{Z��!������}��)�G��.rrr�S��''7�J@<�^B���R xJ�����SQT*��S$�]DRM�E��������*��~O���<�P>5U�)4��b����u��S������S`Z1�{R��e�4G	���{��M,�^ɫ}���q�k��7���ݫQj�'C�nl<�b�dDj��G<��ݫIj��x=�ȀI����M�w�&�U�3M�e�Gx��xʍ��W�z��]1WUU8�<U0���S���)�]1QU7`��$�l�^y��y�(��P�ϓ~���)�ݼ�Be��Q��.�ȤF�x�躂�h��%͘�x
Nm��h�}�|������~O�c�O�J�/�����8�){r�Ӻ/����;����@�=�����ֈ
�RNSG�~y
�J��z*H_<}?�5�� =��)4.�����I��!�'��O)W|iݓҫ`�S�@QP��s��)1�`KU#�g�&'xNu�L��Ԧ����� ��=���m/r#Nn�=��~M��Xr������m'��o�#m�Q�1���#�9r���H�~E�1�CC|�r�vO�F�� ��^��
<��zOA�;L�)}��iy���A%<�o'�k(Kͩ�� ��p��<��j�Hճ��Լ�\�UܢF��+%��!u����L��3�1G�)w�Sy���z���)��8
�S���4�S�B��C��ZP��/A'���}MD������pO��r�������<7p�^��Co�*6���~nt��;�� N�N�(;�Џ����E�S};��R���_[C�
5����� ��]9�O�E)�(cG�/+����0}i��g}&�d,*��,��<<�e��P5��J�Fq��P�E>V:� ����SD%�M�3��Sa��D���Gz=i��w�}������N�=�y�-%�W#��֑fO���|�vO�P�������B�x���O���w�h�3����*��_�yj����nP�	�2<�]x2T*��S5�~Oi<*�I_����,�SUPO1_2T�;OUA�R��2�[���<�{J�-�~OEULŰR�O
񔺧*	��B�4�{UT��WP��Q�y��)pNNUyK��o�K�Gr��D����)ɪ��E�_*.��'ry_�7j��S}��Z��R��tz*./�,�oh������^�m����j�T���������pGx���V���������)�w��H+�k����W��d��1p��0<��>w��<U��ܝ(UH�fU�����T�O�qD�;O��tz
������Y
�SyO(�RO�4"��`���p<��K��-*�+��(t/O
�}
~_җz<�nJ���<D5 O�7�^�.�I�xJ2���S=U�)t]����sb#2���|C�T�y8/�Ut�?�<%���2���x�^��T�x�%E�SZ=τO�}�yj�/���U�f*\��1��T>OeL����y
��/nb�)�9�,5XO�^�9*���M�xW/��L����5��Wi�=�'�R�Y���SY�^�S���ry�o���}���{Jn)ηo�T�%J��t��U�=/�=�q�tV9O����O�8��_t���!�`D0����ڝz�����TX��D�KO>�s���UH��	y
�r�������y2�'��;�R��q=��S�~�����S�OM�)0T�qhx�2���)�S�QՅ<��^�2��Rh�ꯧ���o����{i�T�z�^.�3����~�+�)��T��$}��`��S��
�S�;���麖�y�I>8q��T��)�S����m�c:E�cO��Z�ޒ�O�ONP���z�)�/��f��x<��y�G�uO��}�d��yJ��4�[���Їb��x���דדt��&C�YHi!?�Qt�"^
O�=��iu�G�C����_��g�����tx��<�'�LFQ1��邧x��T�4/������,�ј�a��h����S}�23|~��	�K�4Oͳ�`p�H�T-w]#��2DUȓ��3����a���R	1���)K���L}��ZU��Y�I�$���-PBOY>OZ%h0&�8 �'�cA�F���O��{�7\~O��@��S8�O�ՕZ4Aŗ�D���*�'}��$�Í������ROaR��QꞒt�����,FU��SI����/UO`!m$��W�Q���R�H�$���QJ��@SI�ɫG���F)����,�IO��*�;cX�'H��MRO��M��jF2O��W��y*��B�0��	���S��8}z�5�Q��
H^�y˚ٍ��E�MTZI�G���E�'�`k�Q"�TēoD�?�!�!�������7�D/������2"&�`�5oe��6	��{2}�ȫ���.Sz�dġO�M\M����OU��i��r���	j��T�S�M��)<%"m�*[���P�ty
T�OO�!����<�rX�T����A�<y�SU�6O�D"���������<�.��{��G�\�J<��x
T|�)��`H�S��<x*(x�ʜ�+Mܗz��"��ܲ&�S�[��	<�A�<�$���<+�AP[<e����O�I�\Yv�5��t�-��hz</��*�ψ~f��G/	|y=��%�x�"�m��P�~n�D��4a,cĵ��ri*	_�
<U��ʫ%�(}aX�=H�Su�ZJ�2�%�x�"ڿ�RWͫ���q��"OF���R�!�c�[�uрZ�)��"ݘ*	���O7�ʲjEU�S��Ǟ�NT�xҿ@$[��S�+&�O��Z�=����T^O��Rn��S`�IՈJ|U˴x�o�]<���!]18�Z�<�m@�U�٭O!=��=U&��J
Ó�GV�DUx�6
�j�t����S%�I���j<P5L]��ĆQ煩RMU�)\a6,S��,uē1�ak@�\�x�'�:�*�T�d"f��c��)�r��d�ݤax�5%�S}��S��ZX
Q��m@K�u���%�ʘ���x�ZZ
�_�UaW��:͒ࢿ�x*��w��	��[�l/r������S�H�Mq^X�-����@��	���i�dH�ZU
S1Q^OlybEy|`E2rc�:��ȇ��8�4K���]��}<�a��@
a0�Y��<y"k��=�1�O�$��V
KS��r����xx�s0���z|`"O��*_����<J5OD<�'�������B�z|`<tS
Z�B7��.�s*A货�;��8�Vp6"�h������{˳�/��T�<C��SLr�߭�U�W[�ζ)}�M"qq����&+�����4elh��S���R{���)	�R�Oj_�}��1e�S����3���<"&N�T��Q�
/b	S�x
�)�2d���x240Q�uaz�'*�K�)2������τ�{*��%?!M?����;O�꽯������s-�R�9o��D�^�J(R��"+�{l���"�SxKWs1yyH�'2��ƸI_J�tQ���E�-ƅ~���'"��7�f��7�{2$�Ӧz�##'܆�o]?�wQ���>o�N�j����+���S�]C"�SxL�G9CQ��'CXU(M���&�#�7������)���9K��T���"����EWi�`��)6��*��y
�V��\�N�&�T��c艸'*7��������OD�҃.�x��������֊h�t���ދ"x"hL���*y�7�����,�~Ox��)ppC�2�3�:�&N7�A�'�Т/�Sd��!��VH�����K4u\+q��%}�lD���xJ�=�;J�G�_�s
�������y#���W�S��x��5��5�'�����y��(]SM�)C h<�����DM�P=ɨ���@,��g�tj�<�
�S�<Pq �����x��'*$u�T7OTH$��n���	�
�D�<�
�@}2��x��'�z$|u�@-�h�4��ݾ�����������(�"���Ɔ6�Y�
��gg�=�m��b�����������q�)�
��F�Oa����?𫱋�OM�kM�v�[�h�!�5=m�L�3�������e�I&?2��LWZW���-�m�u���ݦԥ�S�U�^��Aj����$��L��զ���M󘹦/�٦��ӓ��k짦L��t����ɴ}h�m7��&���2L�8ϸc\���q���p���|�q����Z
Ƶ���.�����%��>����T,|:	MC��M�������^������v~��?D6N�>����qR���,�
���b6�Ѷ�m_bm����x�m.�ZW��{�~�,q�W�he��>�p��Pdn%�7wv��	M,�s-݅�n�C䕂��@�O� ��^'��
��˵��Z��R���Z1Qw��[�.���^a.�G���[Ƽ),gz������'q?�؈5ن���fv��av�};߾��c�J쏡q�|t�݋�e� �V����/���"˽'z�%��C�^�o���H��g��sť�[�"�q�0L�+<��s��3h�0
=#B����F#��X�t����C7�>��N�*�J�Y~�6=?�������^�s#�G��9+�E�G�uF�¹��
\T�.@8j��BW�wخ�f�
��\�^dڡ{�Nh}R�S��M�Aj����>K�;Lei�
����N�:��Rװ{���!�]�a�Ƭ%{0�{�m�������������XO�^��A�D~`���k9I�6�k-1ﳮ5��N��V������Qc��У���q�=�}���3�.z%�!Mi��w����axk�l;��J��6P[�שM�L}�u�6c��۱~�mإ֯��ȍ���X�/Kv���ei��j�b{�7�o�����=��L|
������l����w�����y�w��0v��oƺ���Z|O�J�wb�v�7lf��c]�)�����؃��D�Z���z�6ˍE��^�ވ���������F��-���QF�Nm5��6�FS�L6j�ɫ��
ZlK�f�>z���ɬ45e�h���fL�s�V`��5�&q�c7�7b$�3�}�����~�[���^�<�l8=M����b�c��A�%�/��+��6pYZ���p'L-��&O`O��c��&�ƞDy�p6����F�S�'�iأ�iصd>���<v�x�I�iu�F��_J�������֎8A]�w�;�=��������x'��A������Sm��Tk���ߢ��OSg���=g��$ʈ;�S�������ԗ�j����I�����a��N�C��ݟ��J����iT����,*[I���M�h�|������Tl�@��4[J=�m�^��Q���Z9��7��X{�jl��n�	�|l�uf�~���t��X�K���;�{���<� �g���_Aޏ�>��g�/�ދ���Oዬ���zϵ%~���(�x���L}��H��S?����8g݃��n�&���M���
x�&|;���KL n#k���6�7ٟ�C�J\c�B<f}W++���
b�e�y���6����K\I�m��hl�ԒiFf���ͫ���w�D����x�;I|��D��_'>�F���/�0b3�x�K��#M��D���M����h�r�����2�Vz��/3亙�E��3��ӗ�'Q���Z��{�]�M-]�{�.�>Z������ͣ��Ϛ�k^d���k�Esj��/�6���9�m^���c���$?ռ�_m�򙖎|s�_��6���%��E��Z�Q�-m�-,�[ZX֘[Xr�oi9��oY�Q�o�Y����
1��/�l� r�E�ζS�,��W-�3ŖK��r���/����Y�g,#��$�]D�Fȳ؋�]tru99�z	9�����҅�l�Bƻ����ɯ���?1��Id1��n~���e2y�u29��LN��"3O�_��6��dW~i�'�����*�O�Y�������a�E�(9���jgYk��9��Z^�N�U��K�!�W��i�s�	�htܺ��ƚ˶�^gS=���0�^
��>�GR��w������A?R�Q?:��I�L����V���I���FoG�:���|���������~a7�ŞZ�G��lg��גm�/���C#�-HeWr��q>��P�ٷE;���U�,�f��}'�����d�ʎ?b�>a��'l3�c�]�
;uFOi�f�"����F����=���N+-a�}%NC�X��
��;?���\T���a�
5�:r��O���?����+�ĳ���0�jQ�	
w���狹����o��o��7�8������1��?ŷ���'��_-�ŷ{
����0�W�;xY+�����/����\a%2S�S�X�g~}��B��&���Y��Y���`���jb��C����$�ı��֘���1~��S�
�s���(��s�|�8�[����D�!����Cx3�/�����@��4ډ�f��?b	"m%��qb?z���"f�C��/F>���8���4_��#.� �pW�q��|n,ޕ��D#���c��ٳ�V�Y���F�K��t�1>���Fy�d��Q�_��G�`n�ێ�c�c�g�+l!:�Y�����	>e�=�2�Z��A[���
�6�1Kaػ�=L��9o����Tv(�����l3�)[��e~��3��i��=�����Ϙ������2�_����^H����L	N1��ә�X1�c*��N-��Ydr1o�j9�Vf��sf�i({��	���}n��nӧ�'Sw�OS7�i7���~�����7e$Zҙ����ݭ��&�)Z�7Y�$��_�,q�T�d+�4��?h���^"h��h�b�X]��?�ވ��Eu���n��:9�9�\�Zs��|�jl>���Ļ�|���މ��ϸ�I+<�o���@�������Z�i�Ђ,"�Zg����ȓ��z:�|��&;�3���G��+�|.���ϣ��a�
?3?7�s�!�#��߶����,o��]c'�b�H��}"r�'#��7"�<k�c�����h���q�gnw�f:;��/s�Eww�@uv��N�G������Z＋j-��VHwS�|7��wQ�<��P�O�Q���+K��
��k��VP3]�?\%ToueR�R-��P�j����T��ۨ��{�]ǫt�����Ʒd�oż�ݾ<�f������	���ۘ.��1���i2~)3#�Y���~��I5���u��*v����@]�ݟ�~�h�����V�=v����rv�:�]�2��؅�M���Q�b���WՍ�r�׮�L׭�Z�a�)�)Pp�7�V���)�C�c�dwI���L�i�v[Ⱦ#���)岬t#����ڜ�b�9&���/���㮢I��H�wA�F�R�v��9���L���<�XO?���Z�XDe:��⼓n��tt�g,�,�U�v���=��������`.�J*�n��-�ۙV�Ĝ/M`�^��R	c��b��̑�03��L�m�)�j'?���6�>��܏9_���n[j���&�K��Ve�-�@/������e.��E�ר�FU�����������*��qz�z�:��;��o����;���}3�2��$;�^�=�~'�zf���1u�Y������<H�������Rz��ʼ���9�!�V�|��v'�A\�������G���I��W�#��������U���bk۷�[��>޶���m��F�*� �"�-������ٟv\a���~����%��~�y��{�4�go�1�d�7΋�����1�s�Q�0��E���7]�:���:W��K�\�i�����]Z�N*r)�t׫ғ�%ZyG���J�*�H���0�\i��)͐�H�e$�/�&�rO��{ȸ|�<L~^~H~Q.����ur����C�G~C"�M�,崴U9#mS���+�j�q�Ve�c�����}�2ԶAyV|_Y#�U���Wf����t�+n����)/p;�ۦH�vE�+���Sp���1y$(�Za�Q�q4H��:������|-rF���-�Qh��>4�v�/F/��Pg��'<��F"A���Ǹ)Bw���XxOxExZ���������bGq��E�]4��~�-���~��N
o��b�ֺgqO���	�]���8A.�!>'��O�ۅ���?)�܏��_�t�m6���!ʶ��l���0��O�lω�ly⇶�G����'�n[3n��:m{�i������v�
�6�-��b�}��F+��=�uv������h��޷�N��B}s�
�������
�9��:ܶ�Ŷa����َ��k�s�9\��1��:��f:�E��%���t�s�/�|[g7�����\��v�s���<j{ʹ���s�8���tZ�ǝ�q���d�kh��;�J��5���ϓ.[JCm�����R[G{���2	�zI�[�M�`i�4Vj/a�ώ1�A�hi�m�dGK�QR?J�ɍ��py�gh��},r�J����/�6��m�t�}�$9�Io8�J9ڻ�+���@�<�y�4R~�9T~�1LcX��6\�GxX����rCe/�����!o�>�G���]�<�C�+�9ӱM���,��6���Y�nV�����1TYm�L�=��V��+��+Wh�?B��?����4~��˿�L�n_��\��MX��"�W��({l���ѱD�͹T9OZ�ke��H�+-TH�}R�2W�*c����[\K�\M��=�,q�q��B�F�cN�Z��@�޹U=�ܢ����M�C��Q��tX-vQ�8�R/s|����.�S;�~V�R�W;�V�=m�_�U.B���XX�rMՉ�du�pF="���H�.�
aZ�������[٪P�}�0/�(�I�F�p�3����u�-���-�����.�^ǝVGrkԶ�n�S�����Շ�tU�Iu���A����C� ��uzÅP��37�%p)S���\n��.��V�)�
�X�l�`n�+�;Gm�P���'�����w�6�(K���"~��V���oR~ю{��XyC\��mT�:6*g��c�K�Y��R+�YR[���6���\��Z���-]˄�.Ah�BBsW#�FWk�3.��q��Ow����]+���\]�����������n�B[k�Z[�+O��u�x��@X�G�[(S�
We����g7e{����������E�J��F$�V���#�i~�z����#U#G�;�#�"�M]�z��P/�%�S-<�6��/��ċ\=l��l�\K��F9��uΏ��΋e�����W�+��NZK�Iro)W�W{����xy����Y�M�u�J����m�`��|�|@�V�L�Z�-_.�+7�w�m�W�c+�.)rOi��^,�I䳤�ZY(w��ȍ%�|�4JۃG�ҋr��BkKVj�R��R�Pn��/T��V���*��q�m�te��T�i�&��<�d���<��2[�BY �T��c�U�4e�<E�@~Z�,?�,�	�|��O�[�VnRnV:k�+HY-��,�]�v�ye�<O)��+�r�rD~I�D.P��/*'�B����y��+V�i�Ӷ�)��m���B�ey�R$oW8y�2J>�<(�Q9�?���T��B�m��s���N���~�}�葆�2
m�ǻ�w��
�˺4��ܬ�k��N�_��p�4�

```

### What version of an integration has described problem?

v1.6.0

### What was the last working version of an integration?

--

### What vacuum model do you have problems with?

Roborock.V2 (S5)

### Please firmware installed on your Vacuum.

002034 (MiIO) / v0.10.10 (valetudo RE)

### What type of platform you use?

ARM (Raspberry Pi, Odroid, etc.) > 4GB

### What version of Home Assistant do you use?

2024.3.3

### What type of installation are you running?

Home Assistant OS

**Comment by bkueppers** [comment]
Oh, just forgot: Valetudo mapper (https://github.com/rand256/valetudo-mapper) works, but does not give me the rooms (which is what I should expect from what I understand about that tool...)

**Comment by sca075** [comment]
@bkueppers thanks for the report. I'm currently working to correct some functions. The log is showing the map data (binary format) that I forgot to remove before. 
The room should be configured in the vacuum. I use an S5 too to test the camera extracted the rooms. I will check the logs you provided and try to get back you as soon.

**Comment by sca075** [comment]
@bkueppers in your logs there should be:
```
rockrobo: Received vacuum destinations: {"spots":[],"zones":[],"rooms":[{"name":"#1","id":1},{"name":"#16","id":16},{"name":"#17","id":17},{"name":"#18","id":18},{"name":"#2","id":2}],"updated":1712048518418}
```
![Screenshot 2024-04-03 at 01 20 49](https://github.com/sca075/valetudo_vacuum_camera/assets/82227818/6fc856dc-f3f1-4578-9ac2-9f64610b8da7)

Then the rooms should be display. I will try to release a new version tomorrow, hopefully the issue with the errors you provided should be gone by then ;) Is the map display?

**Comment by bkueppers** [comment]
In which log would I find that? 

No, the map is not drawn. Only a grey picture is displayed.

**Comment by sca075** [comment]
> In which log would I find that?
> 
> No, the map is not drawn. Only a grey picture is displayed.

@bkueppers probably I found the issue.. please check if the auto zoom is enable, if so there is an issue with the cropping data that also in my test generate the grey map.

**Comment by sca075** [comment]
@bkueppers I found the problem in the connector.py I will soon release the version that should fix this issue.

**Comment by bkueppers** [comment]
Thank you very much, being able to use your camera with support for the rooms is a considerable improvement for my setup :-)

**Comment by sca075** [comment]
@bkueppers when it will work, please consider to start this project it helps really a lot me :)

**Comment by bkueppers** [comment]
Sadly, it still does not work for me. I've purged the integration and completely re-installed it, but still I get the error with index out of range. 

```
Traceback (most recent call last):
  File "/config/custom_components/valetudo_vacuum_camera/valetudo/rand256/image_handler.py", line 458, in get_image_from_rrm
    self.room_propriety = await self.get_rooms_attributes(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/valetudo_vacuum_camera/valetudo/rand256/image_handler.py", line 615, in get_rooms_attributes
    self.room_propriety = self.extract_room_properties(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/valetudo_vacuum_camera/valetudo/rand256/image_handler.py", line 269, in extract_room_properties
    coordinates[0].pop()
    ~~~~~~~~~~~^^^
IndexError: list index out of range
```

Anything that I can help with to fix this error? Provide specific information, e.g.?

**Comment by bkueppers** [comment]
By the way: How _should_ look the MQTT topic? Which messages are you parsing there?

**Comment by sca075** [comment]
@bkueppers probably this can be the issue in your case:
```python
      for zone in zones_data:
            zone_name = zone.get("name")
            coordinates = zone.get("coordinates")
            coordinates[0].pop()
            x1, y1, x2, y2 = coordinates[0]
            if coordinates:
                zone_properties[zone_name] = {
                    "zones": coordinates,
                    "name": zone_name,
                    "x": ((x1 + x2) // 2),
                    "y": ((y1 + y2) // 2),
                }
                id_count += 1
        id_count = 1
        for point in points_data:
            point_name = point.get("name")
            coordinates = point.get("coordinates")
            x1, y1 = coordinates
            if coordinates:
                point_properties[id_count] = {
                    "position": coordinates,
                    "name": point_name,
                    "x": x1,
                    "y": y1,
                }
                id_count += 1
```
line 266 to 291 in valetudo/rand256/image_handler.py 
if you will provide the camera logs I can check and eventually correct it. You can temporally delete those lines if you feel confident to do it to test if this solve the issue. Somehow the json data are translated else there should be an error on the parser, it looks that "destinations" topic anyhow isn't including the zones or points.

**Comment by bkueppers** [comment]
I will test later if commenting out the lines will fix the thing. 

Maybe a stupid question: If I enabled debug logging, how Do I actually access the log file? Because the log viewer does not seem to give me everything,...

**Comment by sca075** [comment]
@bkueppers okay I found the problem.. this is happening because there are actually no coordinates on the zones. Received vacuum destinations: {"spots":[],"zones":[{"name":"Küche","coordinates":***[]***}]...}
If you edit the zone in the robot.
![Screenshot 2024-04-03 at 15 49 41](https://github.com/sca075/valetudo_vacuum_camera/assets/82227818/883b0360-6123-482b-acb5-be2e73417cb9)
![Screenshot 2024-04-03 at 15 51 15](https://github.com/sca075/valetudo_vacuum_camera/assets/82227818/04e89a08-aa7a-4ca1-9764-b32bf1d65439)

then the camera should load correctly. 
Please give it a try and let me know if it works ;)
Of course at his point I keep this open till the next release to fix this potential issue. Thanks a lot for the help understanding were was the problem, really appreciated.
Have a nice evening.

**Comment by sca075** [comment]
Fix will be released on 2024.04.1 the next week. I just changed the code to avoid when the coordinates are an empty list.    
```
ccess_token: 905425d8b0ed68273d6dd29ceaa16797d9f1b521f6a6e4c32f5fb849ae244385
model_name: Valetudo Vacuums
brand: Valetudo Vacuum Camera
friendly_name: vacuum Camera
vacuum_battery: 100%
vacuum_position: 
x: 25650
'y': 25800
angle: 536
in_room: '#16'

vacuum_topic: valetudo/rockrobo
vacuum_status: docked
json_data: Success
vacuum_json_id: 56bbf2ee-b11f-4a08-b912-4f9d8ef014ce
calibration_points: 
- vacuum:
    x: 33020
    'y': 28340
  map:
    x: 0
    'y': 0
- vacuum:
    x: 13820
    'y': 28340
  map:
    x: 1594
    'y': 0
- vacuum:
    x: 13820
    'y': 17750
  map:
    x: 1594
    'y': 896
- vacuum:
    x: 33020
    'y': 17750
  map:
    x: 0
    'y': 896

snapshot: false
snapshot_path: /local/snapshot_rockrobo.png
entity_picture: /api/camera_proxy/camera.vacuum_camera?token=905425d8b0ed68273d6dd29ceaa16797d9f1b521f6a6e4c32f5fb849ae244385
supported_features: 1
rooms: 
'1':
  number: 1
  outline:
    - - 22100
      - 27250
    - - 17650
      - 27250
    - - 17650
      - 24150
    - - 22100
      - 24150
  name: '#1'
  x: 19875
  'y': 25700
'2':
  number: 2
  outline:
    - - 27250
      - 23050
    - - 24950
      - 23050
    - - 24950
      - 21950
    - - 27250
      - 21950
  name: '#2'
  x: 26100
  'y': 22500
'16':
  number: 16
  outline:
    - - 27950
      - 27200
    - - 22250
      - 27200
    - - 22250
      - 23100
    - - 27950
      - 23100
  name: '#16'
  x: 25100
  'y': 25150
'17':
  number: 17
  outline:
    - - 22200
      - 24500
    - - 20550
      - 24500
    - - 20550
      - 21700
    - - 22200
      - 21700
  name: '#17'
  x: 21375
  'y': 23100
'18':
  number: 18
  outline:
    - - 23500
      - 23500
    - - 21700
      - 23500
    - - 21700
      - 21550
    - - 23500
      - 21550
  name: '#18'
  x: 22600
  'y': 22525

zones: 
test:
  zones:
    - - 24000
      - 24000
      - 25058
      - 24787
  name: test
  x: 24529
  'y': 24393

``` 
vacuum destinations: {"spots":[],"zones":[{"name":"kitchen","coordinates":[]},{"name":"test","coordinates":[[24000,24000,25058,24787,1]]}]                                                                                                                                                             The kitchen coordinates will be then skipped.                                                                                                                                             If you will start this project will be really appriciated.
